spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -328,4 +367,14 @@ private static boolean isSupportedInPredicate(Predicate predicate) {
       return Arrays.stream(predicate.children()).skip(1).allMatch(SparkV2Filters::isLiteral);
     }
   }
+
+  private static class PredicateChildren<T> {
```

Comment:
I don't think it should be necessary to create a custom class for this. Can you use `Pair` instead?

Reply:
Changed to Pair.